android/app/src/main/java/com/msupplycoldchain/SelfSignedCertClientFactory.java

```diff
@@ -0,0 +1,121 @@
+package com.msupplycoldchain;
+
+import android.util.Log;
+
+import com.facebook.react.modules.network.OkHttpClientFactory;
+import com.facebook.react.modules.network.ReactCookieJarContainer;
+
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class SelfSignedCertClientFactory implements OkHttpClientFactory {
+    private static final String TAG = "OkHttpClientFactory";
+    private boolean isPrivate = false;
```

Comment:
I'm a little confused about how this works with a class level variable? I guess it will be ok though given we're only going to a single URL...

Reply:
that was my thinking - we only configure the app with a single URL. it is however set on every request before the SSL certificate is validated